packages/react-reconciler/src/ReactFiberHooks.js

```diff
@@ -2187,7 +2187,7 @@ function mountActionState<S, P>(
 
   // Pending state. This is used to store the pending state of the action.
   // Tracked optimistically, like a transition pending state.
-  const pendingStateHook = mountStateImpl((false: Thenable<boolean> | boolean));
+  const pendingStateHook = mountStateImpl((false: boolean));
```

Comment:
The previous code matches how we set pending throughout this file. Why is this the only place where it isn't a thenable?

Reply:
I understand the desire for consistency, but adding the Thenable type to pendingState when it's not actually used as a Thenable value seems unnecessary. It could make the code harder to understand. I propose we prioritize clarity by matching the type definition to how pendingState is actually used.

Reply:
What about handling the `pendingState` inside `updateActionStateImpl` like this?
```ts
// updateActionStateImpl
const [pendingState] = updateState(false);
const isPending =
  typeof pendingState === 'boolean'
    ? pendingState
    : useThenable(pendingState);
```

Reply:
Is it not used like this in the other cases?

Reply:
In the case of `useTransition`, a thenable value is indeed passed through `chainThenableValue`, as shown in the 
code : https://github.com/facebook/react/blob/190cc990e01e5131a6b26f1a3212444cebd794e2/packages/react-reconciler/src/ReactFiberHooks.js#L2897-L2901
```ts
const thenableForFinishedState = chainThenableValue(thenable, finishedState);
dispatchSetState(fiber, queue, (thenableForFinishedState: any));
```

However, in the `useActionState` case, specifically in the `dispatchActionState` function, `setPendingState` is always called with a boolean value `true`:

https://github.com/facebook/react/blob/190cc990e01e5131a6b26f1a3212444cebd794e2/packages/react-reconciler/src/ReactFiberHooks.js#L2033
```ts
function runActionStateAction<S, P>(
//...
  setPendingState(true);
```
In `useActionState`, `pendingState` is consistently treated as a boolean value, whereas in `useTransition`, it has the flexibility to be either a boolean or a thenable.